### PR TITLE
[aws-sdk-cpp] package files for building SDK plugins

### PR DESCRIPTION
**Pull request by dvirtz**
Specify library name and version:  **aws-sdk-cpp**

See https://github.com/conan-io/conan-center-index/pull/7407#discussion_r731908259

Enable building AWS SDK "plugins" (as done for example by AWS CDI SDK) by packaging build files to be used by those plugins

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/aws-sdk-cpp/all/conanfile.py [hunk 1]
```diff
@@ -398,6 +402,65 @@ def build(self):
...
+            self.copy(file, src=self._source_subfolder, dst=self._res_folder)
+            tools.replace_in_file(os.path.join(self.package_folder, self._res_folder, file), "CMAKE_CURRENT_SOURCE_DIR", "AWS_NATIVE_SDK_ROOT", strict=False)
+
+        # avoid getting error from hook
+        with tools.chdir(os.path.join(self.package_folder, self._res_folder)):
+            rename(self, os.path.join("toolchains", "cmakeProjectConfig.cmake"), os.path.join("toolchains", "cmakeProjectConf.cmake"))
+            tools.replace_in_file(os.path.join("cmake", "utilities.cmake"), "cmakeProjectConfig.cmake", "cmakeProjectConf.cmake")
+
+        # create a cmake module to load the files above
+        contents = textwrap.dedent("""
```
Comment (madebr) [inline recipes/aws-sdk-cpp/all/conanfile.py:428]: I think you copied this from `cmake/AWSSDKConfig.cmake`?
I believe this contains potential version depdendent logic.
Can we move this to a patch, and copy that file here instead?

Comment (dvirtz) [reply recipes/aws-sdk-cpp/all/conanfile.py:428]: I tried to have the minimum amount of CMake code required to build a single SDK.  
it's copied from different parts of the CMake code. That specific file is actually not seemed to be used anymore by the package.

Comment (dvirtz) [reply recipes/aws-sdk-cpp/all/conanfile.py:428]: I'm not sure how to achieve that with a patch

Comment (madebr) [reply recipes/aws-sdk-cpp/all/conanfile.py:428]: The following patch creates a file containing 3 lines:
```patch
--- /dev/null
+++ some_file
@@ -0,0 +1,3 @@
+line1
+line2
+line3
```
Using the same format, you can create a patch for a longer file.

**Comment by dvirtz** [comment]
need to wait for dependencies to be updated due to #7763

**Comment by dvirtz** [comment]
@toge do you know what dependency versions I put for the CI to find them?

**Comment by dvirtz** [comment]
@madebr @uilianries can you please re-review?

File: recipes/aws-sdk-cpp/all/conanfile.py [hunk 2]
```diff
@@ -340,13 +344,13 @@ def validate(self):
...
         if self._use_aws_crt_cpp:
             self.requires("aws-c-cal/0.5.12")
-            self.requires("aws-c-io/0.10.9")
+            self.requires("aws-c-io/0.10.13")
             self.requires("aws-crt-cpp/0.14.3")
         else:
-            self.requires("aws-c-event-stream/0.1.5")
+            self.requires("aws-c-event-stream/0.2.7")
         if self.settings.os != "Windows":
             self.requires("openssl/1.1.1l")
```
Comment (prince-chrismc) [inline recipes/aws-sdk-cpp/all/conanfile.py:355]:
```suggestion
            self.requires("openssl/1.1.1m")
```

**Review by prince-chrismc — APPROVED** [review]
LTGM

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
Please do not force push 🙏 GitHub forces us to restart the review which is not fun!

**Review by danimtb — APPROVED** [review]

**Comment by prince-chrismc** [comment]
@dvirtz You have merge conflicts !

**Comment by prince-chrismc** [comment]
Please add the recipe in the title of the PR pretty please 🙏

**Review by SSE4 — APPROVED** [review]

**Comment by uilianries** [comment]
@dvirtz please, solve these git conflicts when possible.

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Comment by prince-chrismc** [comment]
@ericLemanissier I love the linting ❤️

**Review by SSE4 — APPROVED** [review]

**Pull request merged**